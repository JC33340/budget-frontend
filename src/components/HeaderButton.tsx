import { NavLink } from 'react-router-dom';

type HeaderButtonType = {
    href: string;
    text: string;
};

const HeaderButton = ({ href, text }: HeaderButtonType) => {
    const css =
        'text-lg font-semibold p-2 rounded-md hover:bg-white hover:text-blue transition-all ease-in-out ';

    return (
        <NavLink
            className={({ isActive }) =>
                isActive ? `${css} bg-white text-blue` : `${css} text-white`
            }
            to={href}
        >
            {text}
        </NavLink>
    );
};

export default HeaderButton;
