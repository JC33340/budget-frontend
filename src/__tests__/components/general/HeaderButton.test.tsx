import HeaderButton from '../../../components/HeaderButton';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

test('snapshot test', () => {
    render(
        <BrowserRouter>
            <HeaderButton href="/test-href" text="test-text" />
        </BrowserRouter>
    );
    const header = screen.getByTestId('headerButton-testid');
    expect(header).toMatchSnapshot();
});

test('testing text and href content', () => {
    render(
        <BrowserRouter>
            <HeaderButton href="/test-href" text="test-text" />
        </BrowserRouter>
    );
    const header = screen.getByTestId('headerButton-testid');
    expect(header).toHaveTextContent('test-text');
    expect(header).toHaveAttribute('href', '/test-href');
});
