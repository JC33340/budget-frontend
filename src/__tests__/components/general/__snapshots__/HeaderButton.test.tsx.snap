// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<a
  class="text-lg font-semibold p-2 rounded-md hover:bg-white hover:text-blue transition-all ease-in-out  text-white"
  data-testid="headerButton-testid"
  href="/test-href"
>
  test-text
</a>
`;
